torch/_refs/nn/functional/__init__.py

```diff
@@ -0,0 +1,48 @@
+import torch
+
+import torch._prims.utils as utils
+from torch._prims.utils import TensorLikeType, TensorLike, NumberType, Number
+import torch._refs as refs
+
+from typing import Optional
+
+all = [
+    "elu",
+]
+
+# elu is implemented specially because it has an alpha argument
+def elu(a: TensorLikeType, alpha: Optional[NumberType] = None, inplace: bool = False):
```

Comment:
Interesting - I was going to ask whether we could share this one (https://github.com/pytorch/pytorch/blob/master/torch/_decomp/decompositions.py#L84), but the ATen op seems to have a couple more configuration options. I wonder why...


Also, is there any particular reason you implemented it with this series of operators instead of mimicking the kernel? This is computing `max(0, a) + min(0, (exp(a) - 1)*alpha)`, while the kernel (and the ATen decomposition) is computing ` a > 0 ? a : (exp(a) - 1) * alpha`. They're mathematically identical, but they don't handle propagation of nans identically if `alpha` is `float('nan')`.

Reply:
It's the mathematical formula in the documentation but @ngimel and I are going to take a look at some NaN propagation principles and tests soon, so we're not wedded to this decomp

Reply:
Yeah we should understand the ATen vs Torch api impedance and how best to address it (and possibly retarget what's written in the tracker spreadsheet) cc @ezyang

Reply:
There are 2 competing mathematical formulas in the documentation https://pytorch.org/docs/master/generated/torch.nn.ELU.html?highlight=elu#torch.nn.ELU

Reply:
Wonderful

Reply:
The paper defines the branch version https://arxiv.org/pdf/1511.07289.pdf, moreover, explicit grad formula given in the paper makes the branch version the only one correct.

Reply:
In general, I've been trying to default to "whatever the ATen implementation does".

Reply:
re the extra knobs, it appears to just be a holdout from the TH days. It looks like they might be dead code and can just be removed. cc @gchanan